src/backend/api/Fusion.Resources.Logic/Requests/Commands/ResourceAllocationRequest/JointVenture/Approve.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Fusion.Resources.Database.Entities;
+using Fusion.Resources.Domain.Commands;
+using Fusion.Resources.Domain.Queries;
+using MediatR;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Fusion.Resources.Domain;
+
+namespace Fusion.Resources.Logic.Commands
+{
+    public partial class ResourceAllocationRequest
+    {
+        public partial class JointVenture
+        {
+            public class Approve : TrackableRequest
+            {
+                public Approve(Guid requestId)
+                {
+                    RequestId = requestId;
+                }
+
+                public Guid RequestId { get; }
+
+
+                public class Handler : AsyncRequestHandler<Approve>
+                {
+                    private readonly IMediator mediator;
+
+                    public Handler(IMediator mediator)
+                    {
+                        this.mediator = mediator;
+                    }
+
+                    protected override async Task Handle(Approve request, CancellationToken cancellationToken)
+                    {
+                        var dbRequest = await mediator.Send(new GetResourceAllocationRequestItem(request.RequestId),
+                            CancellationToken.None);
+
+                        if (dbRequest!.Type != QueryResourceAllocationRequest.QueryAllocationRequestType.JointVenture)
+                            throw new NotSupportedException();
```

Comment:
Should include messages in the exception, otherwise you are the designated debugger whenever a notsupportedexception is traced :)